dreammall-earth/dreammall.earth
# feat(backend): delete user via authentik

Motivation
----------
This will synchronize user deletions in Authentik to our own database.

How to test
-----------
1. `docker compose up` (important!)
2. Log into Authentik as admin
3. Create a new user via Directory>>Users
4. Impersonate as this user and change your password
5. Log out and log in as the new user
6. Visit http://localhost:3000 to create a user in our database via JWT
7. Log out and login in as admin
8. Delete the test user in the Authentik admin dashboard under Directory>>Users
9. Backend will receive the webhook and delete the user

It's also deployed at: https://delete-user-via-authentik.git.dreammall.earth/

1. Click on "Signup" and create a new user account
2. Click on the link in your confirmation email
3. Sign out
4. Sign as admin at https://auth.delete-user-via-authentik.git.dreammall.earth/
5. On your terminal, open `k9s` and enter a shell into the `mariadb` pod
6. Run `mysql -u dreammall -p dreammall.earth`
7. Enter the password from `sops infrastructure/helmfile/secrets/dreammall.yaml`
8. Execute `SELECT * FROM User;`
9. You should see your new user account
10. Now delete the user in your admin dashboard
11. Execute `SELECT * FROM User;` again
12. User is gone
